src/lib/models/ConfigModel.ts

```diff
@@ -20,50 +21,51 @@ export interface ConfigModel {
 	toFormattedOutput(format: OutputFormat): string;
 }
 
-export type ConfigContext = {
-	config: CodeAnalyzerConfig;
-	core: CodeAnalyzer;
-	rules: RuleSelection;
-}
-
-export type ConfigModelGeneratorFunction = (relevantEngines: Set<string>, userContext: ConfigContext, defaultContext: ConfigContext) => ConfigModel;
-
 export class AnnotatedConfigModel implements ConfigModel {
-	private readonly relevantEngines: Set<string>;
-	private readonly userContext: ConfigContext;
-	private readonly defaultContext: ConfigContext;
-
-	private constructor(relevantEngines: Set<string>, userContext: ConfigContext, defaultContext: ConfigContext) {
+	private readonly config: CodeAnalyzerConfig; // TODO: It would be nice if we updated the CodeAnalyzer (in our core module) to just return its CodeAnalyzerConfig with a getter so we didn't need to pass it around
+	private readonly codeAnalyzer: CodeAnalyzer;
+	private readonly userRules: RuleSelection;
+	private readonly allDefaultRules: RuleSelection;
```

Comment:
Technically I could have passed in a function that just gives me back the default rule since that's how we are using this. But it was easy enough to just pass this in.

But now we don't need to pass in all the values and config. So no more need for the ConfigContext stuff.